numpy/random/mtrand/mtrand.pyx

```diff
@@ -936,14 +1069,24 @@ cdef class RandomState:
                [3, 2, 2, 0]])
 
         """
-        if high is not None and low >= high:
-            raise ValueError("low >= high")
-
         if high is None:
             high = low
             low = 0
 
-        return self.random_integers(low, high - 1, size)
+        key = np.dtype(dtype).name
+        if not key in _randint_type:
+            raise ValueError('Unrecognized dtype "%s" for randint' % key)
+        lowbnd, highbnd, randfunc = _randint_type[key]
+
+        if low < lowbnd:
+            raise ValueError("low is out of bounds for %s" % key)
+        if high > highbnd:
+            raise ValueError("high is out of bounds for %s" % key)
+        if low >= high:
+            raise ValueError("low >= high")
```

Comment:
This is not within the scope of this PR, but I was thinking that ```randint``` could one day implement something similar to ```numpy.array``` in which it tries to determine the minimum ```dtype``` required to generate numbers in the range requested.  Throwing that out as another idea for future "expansion" FLOABT.